### PR TITLE
dev: Migrate from flake8 to ruff

### PR DESCRIPTION
**Pull request by adigitoleo**
Faster linting and ruff also supports pyproject.toml configuration. Also moves all configuration for other dev tools to pyproject.toml.

### CODE REVIEW

**Review by Patol75 — APPROVED** [review]

**Review by Patol75 — CHANGES_REQUESTED** [review]
The repo address is incorrect (https://beta.ruff.rs/docs/usage/).

**Comment by Patol75** [comment]
So we are getting a few `E501`: up to you if you want to fix or ignore them.
And you probably need to skip `*.*vtu` globally in the YAML.

**Review by Patol75 — APPROVED** [review]

**Comment by adigitoleo** [comment]
Got rid of the line length errors and I think I've put back the vtu exclude now, hopefully no typos, just waiting on the pre-commit job to do its thing again.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -35,20 +35,12 @@ repos:
...
-        args: ["--max-line-length", "88", "--extend-ignore", "E203,E501,E722,W503"]
-        exclude: |  # https://pre-commit.com/#regular-expressions
-            (?x)^(
-                src/initial_implementation/AlternativeFunctions.py|
-                src/pydrex/__init__.py
-            )$
-
-exclude: \.vtu$
+    -   id: ruff
+-   exclude: \.*vtu$
```
Comment (Patol75) [inline .pre-commit-config.yaml:46]: `exclude: \.*vtu$`

**Review by Patol75 — CHANGES_REQUESTED** [review]
Missed that one.

**Comment by Patol75** [comment]
You also need to revert the changes to `src/pydrex/data/steadyflow/corner2d_2cmyr_5e5x1e5.vtu`.

**Review by Patol75 — APPROVED** [review]

**Pull request merged**